src/proc/Vigad.ts

```diff
@@ -52,43 +53,136 @@ export class Vigad {
-     * @param id
-     * @return void
+     * Generates a unique capture area ID.
+     * @returns {string} - The generated capture area ID.
      */
-    public deleteCaptureArea(id: number): void {
-        this.captureAreas.splice(id, 1);
-        this.tesseractHandler.removeWorker();
+    public generateCaptureAreaId(): string {
+        const {
+            numbers,
+            lowercaseLetters,
+            uppercaseLetters,
+            generateValidToken,
+        } = useTokenGenerator();
+
+        let caId: string;
+
+        do {
+            caId = generateValidToken(
+                6,
+                numbers + lowercaseLetters + uppercaseLetters,
+                true,
+                true,
+                false,
+                true,
+                false
+            );
+        } while (this.captureAreaIdExists(caId));
+
+        return caId;
     }
 
     /**
-     * Get a capture area by id
-     * @param id
-     * @return CaptureArea
+     * Check if the capture area ID already exists.
+     * @param {string} caId - The capture area ID to check.
+     * @returns {boolean} - Returns true if the capture area ID exists, false otherwise.
      */
-    public getCaptureArea(id: number): CaptureArea {
-        return this.captureAreas[id];
+    private captureAreaIdExists(caId: string): boolean {
+        return this.captureAreas.some((area) => area.getId() === caId);
+    }
+
+    /**
+     * Renames a capture area by updating its ID.
+     * @param {string} oldId - The old ID of the capture area.
+     * @param {string} newId - The new ID to assign to the capture area.
+     * @returns {boolean} - Returns true if the renaming was successful, false otherwise.
+     */
+    public renameCaptureArea(oldId: string, newId: string): boolean {
+        // Find the index of the capture area object with the oldId
+        const captureAreaIndex = this.captureAreas.findIndex(
+            (area) => area.getId() === oldId
+        );
+
+        // If the capture area with the oldId doesn't exist, return
+        if (captureAreaIndex === -1) {
+            return false;
+        }
+
+        // Check if any other capture area already uses the newId
+        const isIdAlreadyUsed = this.captureAreas.some(
+            (area, index) =>
+                index !== captureAreaIndex && area.getId() === newId
+        );
+
+        // If the newId is already used, return false
+        if (isIdAlreadyUsed) {
+            return false;
+        }
+
+        // Update the id of the capture area
+        this.captureAreas[captureAreaIndex].setId(newId);
+
+        // Return true to indicate successful renaming
+        return true;
+    }
+
+    /**
+     * Delete the capture area with the specified ID.
+     * @param {number} id - The ID of the capture area to delete.
+     * @returns {boolean} - Returns true if the capture area is successfully deleted, false otherwise.
+     */
+    public deleteCaptureArea(id: string): boolean {
```

Comment:
deleting a capture area does not remove the id from upload queue:
![image](https://github.com/VisualGameData/VIGAD/assets/58397976/75fdc106-ab34-42c4-a238-b65e24763608)